modules/ingest-common/src/main/java/org/opensearch/ingest/common/RemoveProcessor.java

```diff
@@ -79,6 +80,25 @@ public IngestDocument execute(IngestDocument document) {
                     throw new IllegalArgumentException("field [" + path + "] doesn't exist");
                 }
             }
+            // cannot remove _index, _version and _version_type.
+            if (path.equals(IngestDocument.Metadata.INDEX.getFieldName())
+                || path.equals(IngestDocument.Metadata.VERSION.getFieldName())
+                || path.equals(IngestDocument.Metadata.VERSION_TYPE.getFieldName())) {
+                throw new IllegalArgumentException("cannot remove metadata field [" + path + "]");
+            }
+            // removing _id is disallowed when there's an external version specified in the request
+            if (path.equals(IngestDocument.Metadata.ID.getFieldName())
+                && !document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class)
```

Comment:
`document.getFieldValue(...`) could return `null` (as per java doc):

```suggestion
                && !Objects.equals(document.getFieldValue(IngestDocument.Metadata.VERSION_TYPE.getFieldName(), String.class), VersionType.toString(VersionType.INTERNAL))
```

Reply:
Thanks, I've changed that.